garak/detectors/base.py

```diff
@@ -164,6 +164,9 @@ def detect(self, attempt: garak.attempt.Attempt) -> List[float | None]:
             # detector_results[non_none_offsets[k]] = norm_result_score # we'll want this when outputs+scores need to align
             detector_results.append(norm_result_score)
 
+        if len(detector_results) != len(all_outputs):
+            raise ValueError("Detector output mismatch - got %s results for %s outputs", len(detector_results), len(all_outputs))
+
```

Comment:
This will be removed from this PR as #1415 provides more comprehensively revision for overall test expectations of detector result counts via unit testing.  As is this change in isolation causes failures for existing tests.